internal/backends/sqlite/metadata/pool/pool.go

```diff
@@ -244,3 +244,34 @@ func (p *Pool) Drop(ctx context.Context, name string) bool {
 
 	return true
 }
+
+// InTransaction wraps the given function f in a SQLite transaction.
+//
+// If f returns an error, the transaction is rolled back.
+//
+// Passed context will be used for transaction.
+// Context cancellation DOES rollback the transaction.
+func InTransaction(ctx context.Context, db *sql.DB, f func(*sql.Tx) error) (err error) {
```

Comment:
If that function is needed at all (and I'm not sure about it), please:
1. Move it to where it is used.
2. Unexport it.
3. Make it handle the logic described in the large comment inside `InTransactionKeep`.

Reply:
Why is it not needed? It seems to me much better than implementing rollbacks for each new function. Implementing the transaction boilerplate in every function might create some places where we don't do rollback that should be done.

Reply:
Please take a look at `TestInTransactionRollback`, I believe it covers everything from the comment inside `InTransactionKeep`.

Reply:
(FWIW: panic in `f` would leak transaction)